### PR TITLE
Adding TestNG and autocreating database using Liquibase in tests

### PR DESCRIPTION
**Pull request by VanDavv**
I managed to upgrade tests DAO root, so now it can automatically create test database using migrations.xml. I also wrote a raw type method, which allows us to get the content of this database in just one line. 
I put it up into TestNG suite.

I'm currently working on changing tests in app module to be run with TestNG, but DAO tests are fixed

### CODE REVIEW

File: backend/app/src/test/java/com/bls/patronage/IntegrationTests.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+public class IntegrationTests {
+
+    private static final String TMP_FILE = createTempFile();
+    private static final String CONFIG_PATH = ResourceHelpers.resourceFilePath("testConfiguration.yml");
+
+    @ClassRule
+    public static final DropwizardAppRule<StudyBoxConfiguration> RULE = new DropwizardAppRule<>(
+            StudyBox.class, CONFIG_PATH,
+            ConfigOverride.config("database.url", "jdbc:h2:" + TMP_FILE));
```
Comment (mlotysz) [inline backend/app/src/test/java/com/bls/patronage/IntegrationTests.java]: Use H2 on memory. It'll speed up us for free.

File: backend/db/src/test/java/com/bls/patronage/dao/DAOTest.java [hunk 2]
```diff
@@ -18,20 +27,30 @@
...
     public void tearDown() throws Exception {
         handle.close();
     }
 
+    private void migrateDatabase() throws LiquibaseException {
+        liquibase = new Liquibase("migrations.xml", new ClassLoaderResourceAccessor(), new JdbcConnection(handle.getConnection()));
+        liquibase.update("");
+    }
+
+    public  <Type> List<Type> getObjectsFromDatabase(Class<Type> typeClass, Class mapperClass, String tableName) throws Exception {
```
Comment (mlotysz) [inline backend/db/src/test/java/com/bls/patronage/dao/DAOTest.java]: It's OK. But I'll do that way:
```
public  <E> List<E> getAllEntities(Class<E> type, Class mapperClass, String tableName)
```

**Comment by mlotysz** [comment]
That's I like the most! We're deleting code which is good sign. Change to H2 in memory configuration and I merge it.

File: backend/app/src/test/java/com/bls/patronage/IntegrationTests.java (hunk 1 above)
Comment (VanDavv) [inline backend/app/src/test/java/com/bls/patronage/IntegrationTests.java]: I'm thinking of removing this class. Doesn't Travis do the integration tests?

Comment (mlotysz) [inline backend/app/src/test/java/com/bls/patronage/IntegrationTests.java]: travis do anything what you tell him to do. :) But what is this class for? I think thaht is base class for DAO test. If it's not connected remove it. Just make sure you control when DB is initialized with initial data.

Comment (VanDavv) [inline backend/app/src/test/java/com/bls/patronage/IntegrationTests.java]: For now, the data loading is in the migrations.xml file. But I'm thinking of making it more dynamic, by adding a generic type method, similar to getObjectsFromDatabase , which will add the data to database, without involving migrations xml

File: backend/db/src/test/java/com/bls/patronage/dao/DAOTest.java (hunk 2 above)
Comment (VanDavv) [inline backend/db/src/test/java/com/bls/patronage/dao/DAOTest.java]: Ok, renamed :)

**Comment by mlotysz** [comment]
Good job! Thanks!

**Pull request merged**